06week/higherOrder.js

```diff
@@ -2,26 +2,135 @@
 
 const assert = require('assert');
 
-function forEach(arr, callback) {
-  // Your code here
+//random array of elements
+
+let array1 = [1,2,3,4,5,6]
+
+//this is my callback function that changes the array in some way, in this case to increase the array
+function incrimenter(num){
+  if(num<4){
+    return (num + 2)
+  } else {
+    return (num * 2)
+  }
 }
 
-function map(arr, callback) {
-  // Your code here
+//This function takes in the array and passes each value into the callback function to increase the array
+function forEach(array, callback) {
+  let changedElements = [];
+  for (let i=0; i < array.length; i++){
+    let converter = incrimenter(array[i]);
+    changedElements.push(converter);
+  }
+  return changedElements;
```

Comment:
The forEach function does not return the callback applied to each element, just executes it on each element.